### ISSUE TITLE
Speaker Regression Loss converge

### ISSUE THREAD

**Issue by Kristopher-Chen** [SongRongLee/mir-svc#1]
Hi, how does the speaker classification loss look like on your side? It diverges in the middle of the training process in my experiment. Any idea about how to balance the speaker classification loss for the classification network(positive) and the main network(negative)?

See the training loss curve for speaker classification below. (My network follows the ideas of speaker classification and pitch regression modules, but not the exact one as your task.)

![image](https://user-images.githubusercontent.com/38187438/147308249-3531443a-c878-40ca-affd-d9a5e6e7b11f.png)

**Comment by SongRongLee** [comment]
Hi @Kristopher-Chen 
I have encountered a similar problem in the research. This means your classifier is not "strong enough" to train against your main network. You can try any of the following:
1. Adjust the learning rate schedulers of main and adversarial networks individually
2. Adjust learning rate of your classifier (in your case, make it larger)
3. Increase the complexity of your classifier
4. Adjust the weights of main and adversarial loss  

These are the methods I can think of at the moment, there are really many things you can try out.  

The following are the loss curves of a successful training of the project for your reference:
![1439A6C0-2431-41CA-BDE4-FBF18888FFE3](https://user-images.githubusercontent.com/17636996/147310634-4a5fa04f-b26f-4ace-a483-9d69535b0ef5.jpeg)

**Comment by Kristopher-Chen** [comment]
@SongRongLee Thank you! I will try to make the classifier more complicated and make the learning rate larger.
